src/apps/js_generic/js_generic.cpp

```diff
@@ -693,99 +697,9 @@ namespace ccfapp
     {
-      const auto local_method = method.substr(method.find_first_not_of('/'));
-
-      const auto scripts = args.tx.get_view(this->network.app_scripts);
-
-      // Try to find script for method
-      // - First try a script called "foo"
-      // - If that fails, try a script called "POST foo"
-      auto handler_script = scripts->get(local_method);
-      if (!handler_script)
-      {
-        const auto verb_prefixed =
-          fmt::format("{} {}", verb.c_str(), local_method);
-        handler_script = scripts->get(verb_prefixed);
-        if (!handler_script)
-        {
-          args.rpc_ctx->set_response_status(HTTP_STATUS_NOT_FOUND);
-          args.rpc_ctx->set_response_body(fmt::format(
-            "No handler script found for method '{}'", verb_prefixed));
-          return;
-        }
-      }
-
-      JSRuntime* rt = JS_NewRuntime();
-      if (rt == nullptr)
-      {
-        throw std::runtime_error("Failed to initialise QuickJS runtime");
-      }
-
-      JS_SetMaxStackSize(rt, 1024 * 1024);
-
-      JSModuleLoaderArg js_module_loader_arg{&this->network, &args.tx};
-      JS_SetModuleLoaderFunc(
-        rt, nullptr, js_module_loader, &js_module_loader_arg);
-
-      JSContext* ctx = JS_NewContext(rt);
-      if (ctx == nullptr)
-      {
-        JS_FreeRuntime(rt);
-        throw std::runtime_error("Failed to initialise QuickJS context");
-      }
-
-      // Register class for KV
-      {
-        auto ret = JS_NewClass(rt, kv_class_id, &kv_class_def);
-        if (ret != 0)
-        {
-          throw std::logic_error(
-            "Failed to register JS class definition for KV");
-        }
-      }
-
-      // Register class for KV map views
-      {
-        auto ret =
-          JS_NewClass(rt, kv_map_view_class_id, &kv_map_view_class_def);
-        if (ret != 0)
-        {
-          throw std::logic_error(
-            "Failed to register JS class definition for KVMap");
-        }
-      }
-
-      // Register class for request body
-      {
-        auto ret = JS_NewClass(rt, body_class_id, &body_class_def);
-        if (ret != 0)
-        {
-          throw std::logic_error(
-            "Failed to register JS class definition for Body");
-        }
-        JSValue body_proto = JS_NewObject(ctx);
-        size_t func_count =
-          sizeof(js_body_proto_funcs) / sizeof(js_body_proto_funcs[0]);
-        JS_SetPropertyFunctionList(
-          ctx, body_proto, js_body_proto_funcs, func_count);
-        JS_SetClassProto(ctx, body_class_id, body_proto);
-      }
-
-      auto global_obj = JS_GetGlobalObject(ctx);
-
-      auto console = JS_NewObject(ctx);
-      JS_SetPropertyStr(ctx, global_obj, "console", console);
-
-      JS_SetPropertyStr(
-        ctx, console, "log", JS_NewCFunction(ctx, ccfapp::js_print, "log", 1));
-
       auto ccf = JS_NewObject(ctx);
-      JS_SetPropertyStr(ctx, global_obj, "ccf", ccf);
 
       JS_SetPropertyStr(
```

Comment:
Since we're exposing a growing amount of custom APIs, we ought to document them. I don't think sphinx-js can help us because it wants to parse javascript, but we can use the Javascript domain to expose them manually: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#the-javascript-domain

Reply:
Can you create an issue for this?